File: app/src/main/java/com/arejas/dashboardofthings/domain/usecases/implementations/ActuatorManagementUseCaseImpl.java
package com.arejas.dashboardofthings.domain.usecases.implementations;

import androidx.lifecycle.LiveData;

import com.arejas.dashboardofthings.data.interfaces.DotRepository;
import com.arejas.dashboardofthings.domain.entities.database.Actuator;
import com.arejas.dashboardofthings.domain.entities.extended.ActuatorExtended;
import com.arejas.dashboardofthings.domain.entities.result.Resource;
import com.arejas.dashboardofthings.domain.usecases.ActuatorManagementUseCase;

import org.jetbrains.annotations.NotNull;

import java.util.List;

public class ActuatorManagementUseCaseImpl implements ActuatorManagementUseCase {

    private final DotRepository repository;

    public ActuatorManagementUseCaseImpl(DotRepository repository) {
        this.repository = repository;
    }

    @Override
    public LiveData<Resource<List<ActuatorExtended>>> getListOfActuators() {
        return repository.getListOfActuators();
    }

    public LiveData<Resource<List<ActuatorExtended>>> getListOfActuatorsMainDashboard() {
        return repository.getListOfActuatorsMainDashboard();
    }

    public LiveData<Resource<List<ActuatorExtended>>> getListOfActuatorsLocated() {
        return repository.getListOfActuatorsLocated();
    }

    @Override
    public LiveData<Resource<ActuatorExtended>> getActuator(@NotNull Integer actuatorId) {
        return repository.getActuator(actuatorId);
    }

    @Override
    public LiveData<Resource> createActuator(@NotNull Actuator actuator) {
        return repository.createActuator(actuator);
    }

    @Override
    public LiveData<Resource> updateActuator(@NotNull Actuator actuator) {
        return repository.updateActuator(actuator);
    }

    @Override
    public LiveData<Resource> deleteActuator(@NotNull Actuator actuator) {
        return repository.deleteActuator(actuator);
    }
}

File: app/src/main/java/com/arejas/dashboardofthings/domain/usecases/implementations/ActuatorManagementUseCaseImpl.kt
package com.arejas.dashboardofthings.domain.usecases.implementations

import androidx.lifecycle.LiveData

import com.arejas.dashboardofthings.data.interfaces.DotRepository
import com.arejas.dashboardofthings.domain.entities.database.Actuator
import com.arejas.dashboardofthings.domain.entities.extended.ActuatorExtended
import com.arejas.dashboardofthings.domain.entities.result.Resource
import com.arejas.dashboardofthings.domain.usecases.ActuatorManagementUseCase

class ActuatorManagementUseCaseImpl(private val repository: DotRepository) :
    ActuatorManagementUseCase {

    override val listOfActuators: LiveData<Resource<List<ActuatorExtended>>>?
        get() = repository.listOfActuators

    override val listOfActuatorsMainDashboard: LiveData<Resource<List<ActuatorExtended>>>?
        get() = repository.listOfActuatorsMainDashboard

    override val listOfActuatorsLocated: LiveData<Resource<List<ActuatorExtended>>>?
        get() = repository.listOfActuatorsLocated

    override fun getActuator(actuatorId: Int): LiveData<Resource<ActuatorExtended>>? {
        return repository.getActuator(actuatorId)
    }

    override fun createActuator(actuator: Actuator): LiveData<Resource<*>> {
        return repository.createActuator(actuator)
    }

    override fun updateActuator(actuator: Actuator): LiveData<Resource<*>> {
        return repository.updateActuator(actuator)
    }

    override fun deleteActuator(actuator: Actuator): LiveData<Resource<*>> {
        return repository.deleteActuator(actuator)
    }
}
